torch/_inductor/codegen/triton.py

```diff
@@ -1314,7 +1314,8 @@ def load(self, name: str, index: sympy.Expr):
                 names = {name}
             last_use = len(names & self.last_usage) > 0
             evict_last = not last_use and ("rmask" in mask or indirect_indexing)
-            ep = ", eviction_policy='evict_last'" if evict_last else ""
+            ep = ", eviction_policy='evict_last'" if evict_last else ", eviction_policy='evict_first'"
```

Comment:
For this, how do the benchmarks look like?

As a side note, I wanted to set the cache modifier to mark it as streaming data (`.cs`) but this is not implemented in triton yet, but I guess this is similar.

Reply:
Actually, adding `.cg` (which is supported by triton) for loads when there is no further use, and for stores when they are outside of a reduction loop should help with kernels that have spilled registers.

Reply:
@lezcano overall, there do seem to be some slight regressions.

I'd like to set `evict_first` *only* if there is another broadcasted load within the same loop. Any thoughts on that and the easiest way to accomplish it?

Reply:
In the scheduler you can find the loads in the current reduction looking at loads between EnableReduction and Disable reduction, and then you have the function `is_broadcast` to tell you whether any of those is going to broadcast, and in this function you have the conditions when a load is going to be lifted outside the reduction loop:
```python
        elif (
            self.inside_reduction
            and not self.persistent_reduction
            and "rmask" not in mask
            and not indirect_indexing
        ):
            # can lift a common load outside of reduction loop
            # One exception is when this is an indirect_load.
```